tools/perf-test-helper.js

```diff
@@ -0,0 +1,45 @@
+import chai from '@open-wc/testing/import-wrappers/chai.js';
+import { fixture } from '@open-wc/testing';
+
+const numComponents = 1000;
+
+export async function runTest(element) {
+	return new Promise((resolve) => {
+
+		let total = 0;
+		let count = 0;
+		window.d2lPerfTestInProgress = true;
+		document.body.addEventListener('d2l-component-perf', (e) => {
+			count++;
+			total += e.detail.value;
+			if (count === numComponents) {
+				const avg = Math.round(total / count);
+				delete window.d2lPerfTestInProgress;
+				resolve(avg);
+			}
+		});
+
+		for (let i = 0; i < numComponents; i++) {
+			fixture(element);
```

Comment:
We render the component 1000 times on the page, then wait for all the events to fire and calculate the average render time.